src/codeu/chat/DerbyStore.java

```diff
@@ -81,18 +82,18 @@ public DerbyStore() {
 			stmt = conn.createStatement();
 			
 			// Create the chat user table
-			stmt.execute("CREATE TABLE " + userTableName + "(id varchar(255), name varchar(255), password varchar(255), creation BIGINT)");
+			stmt.execute("CREATE TABLE " + userTableName + " (userid varchar(255), name varchar(255), password varchar(255), creation BIGINT, PRIMARY KEY (userid))");
```

Comment:
nit: maaaybe some comments explaining what the userid (and other primary keys below) is would be helpful? Also, is name guaranteed to be unique?